hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -182,9 +185,14 @@ private void init(String fileId, String partitionPath, HoodieBaseFile baseFileTo
       // Create Marker file
       createMarkerFile(partitionPath, newFileName);
 
+      Option<Schema.Field> keySchemaFieldID = Option.empty();
```

Comment:
can we move this to writeHandle

Reply:
write handle doesn't track these details today. its all done in derived classes.